ntpd/src/force_sync/algorithm.rs

```diff
@@ -116,8 +150,21 @@ impl<C: NtpClock> TimeSyncController for SingleShotController<C> {
         Ok(())
     }
 
-    fn add_source(&mut self, _id: Self::SourceId) -> Self::SourceController {
-        SingleShotSourceController {
+    fn add_source(&mut self, _id: Self::SourceId) -> Self::NtpSourceController {
+        SingleShotSourceController::<NtpDuration> {
+            delay_type: PhantomData,
+            min_poll_interval: self.min_poll_interval,
+            done: false,
+        }
+    }
+
+    fn add_sock_source(
```

Comment:
bit bikeshedding, but I would prefer a more generic name, as this is likely going to be used for more than just sock sources.